vcd/resource_vcd_lb_app_rule.go

```diff
@@ -0,0 +1,220 @@
+package vcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdLBAppRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdLBAppRuleCreate,
+		Read:   resourceVcdLBAppRuleRead,
+		Update: resourceVcdLBAppRuleUpdate,
+		Delete: resourceVcdLBAppRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdLBAppRuleImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD organization in which the Application Rule is located",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD virtual datacenter in which the Application Rule is located",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the Application Rule is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique Application Rule name",
+			},
+			"script": &schema.Schema{
+				Required: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The script for the application rule. Each line as a separate array element",
+			},
+		},
+	}
+}
+
+func resourceVcdLBAppRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	LBRule, err := getLBAppRuleType(d)
+	if err != nil {
+		return fmt.Errorf("unable to expand load balancer app rule: %s", err)
+	}
+
+	createdPool, err := edgeGateway.CreateLBAppRule(LBRule)
```

Comment:
`CreateLBAppRule(LBRule)` - IMHO would be much better readable as `CreateLBAppRule(name, script)`. as it's now it shows that LBRULE ir complex, but it isn't

Reply:
I could create wrapper, but I would like to keep this consistent and transparent to the struct we have in `types`.
The reason I dislike when we introduce our own "simplified types" for structs is that we end up translating between structs. Then if a new field appears (or we missed some) in the configuration our APIs become deprecated and a new method must be created.
Again if this is a team decision I can change :)
@lvirbalas, @dataclouder, your thoughts about it?

Reply:
One more thought. What if I added more explanation to the type fields we have in Go fashion. That way any modern IDE would show more context of what is expected in the field but would still keep the transparency.

These are mockups comments, not exactly the ones I intend to add.
<img width="1144" alt="image" src="https://user-images.githubusercontent.com/15804230/60948416-8a29d880-a2fb-11e9-82e4-bb65131df3f0.png">

Reply:
> The reason I dislike when we introduce our own "simplified types" for structs is that we end up translating between structs. Then if a new field appears (or we missed some) in the configuration our APIs become deprecated and a new method must be created.
Again if this is a team decision I can change :)

I dislike the simplified types as well, as they add another layer of abstraction.

Reply:
Well, seems we have diff reading and API usage experience. For me as terraform and API reader/user it's easier to pass two string, than Structure with four fields which I have to check which to fill. From Java world there is `rule` if method with 4-5 parameters when refactor to pass object.

Reply:
It may be so. But I have already burnt with such methods in this project. I had to change quite a few details on VM networking just because there were "simplified" fields. The public API gets too many changes this way.

Reply:
In the light of current discussion - what if I add a wrapper for it now and we have both options available?

Reply:
I think you already have this approach for all other previous LB resources. Also, there are many other instances of this approach in EG already. Hence, I suggest leaving this as is in this PR, but  @dataclouder could add this to refactoring consideration together with https://github.com/vmware/go-vcloud-director/issues/211

Reply:
I am OK with @Didainius approach. 
Simplified structures should be used only when: (I) we have more than 3 fields to pass by position, (II) the original structures are not flat, (III) require internal structure manipulation, (IV) and can't be passed to update methods without changes. A clear example is Edge Gateway, where the structure contains a huge amount of details that we may not intend to deal with in every operation. But this structure is manageable as it is.